pkg/lib/interaction/nodes/do_create_user.go

```diff
@@ -79,6 +80,26 @@ func (n *NodeDoCreateUser) GetEffects() ([]interaction.Effect, error) {
 				return err
 			}
 
+			newIdentities := graph.GetUserNewIdentities()
+			ip := httputil.GetIP(ctx.Request, bool(ctx.TrustProxy))
+			isAnonymous := false
+			for _, i := range newIdentities {
+				if i.Type == model.IdentityTypeAnonymous {
+					isAnonymous = true
+				}
+			}
+
+			if !n.BypassRateLimit && isAnonymous {
+				// check the rate limit only before running the effects
+				pass, _, err := ctx.RateLimiter.CheckToken(interaction.SignupAnonymousUserRateLimitBucket(ip))
```

Comment:
Should we follow our convention to call CheckToken in Instantiate?

Reply:
The cases for normal signup and anonymous user rate limit are a bit different.

For normal signup, we called `CheckToken` in `Instantiate` since we want to block the user as soon as we could and user will be blocked in the first step of the auth ui flow. So user could have better UX, it is annoying if user is blocked after completing the whole auth ui flow.

For signup anonymous user check, we use `graph.GetUserNewIdentities` to check if there is any anonymous identity. We need to ensure the node `NodeDoCreateIdentity` has been run, so we need to do the check in the `GetEffects` functions. Also signup anonymous user won't have multiple steps, check it in the `GetEffects` should not have UX problem.

Let me add some comments to the code.